lib/engine.py

```diff
@@ -541,11 +540,12 @@ def format_insert_value(self, value, datatype):
         elif datatype == "char":
             if strvalue.lower() in nulls:
                 return "null"
-            # automatically escape quotes in string fields
-            if hasattr(self.table, "escape_double_quotes") and self.table.escape_double_quotes:
-                strvalue = self.escape_double_quotes(strvalue)
-            if hasattr(self.table, "escape_single_quotes") and self.table.escape_single_quotes:
-                strvalue = self.escape_single_quotes(strvalue)
+            if escape:
+                # automatically escape quotes in string fields
+                if hasattr(self.table, "escape_double_quotes") and self.table.escape_double_quotes:
+                    strvalue = self.escape_double_quotes(strvalue)
+                if hasattr(self.table, "escape_single_quotes") and self.table.escape_single_quotes:
+                    strvalue = self.escape_single_quotes(strvalue)
```

Comment:
@ethanwhite, after we removed the issue of creating auto fields for all tables when using file database engines, the file based engines where escaping some strings. This made the md5sums different between the file based engines (xml, json, csv) and the non file based engines (postgres,mysql, sqlite).

So the file based engines inherit the Engine's `format_insert_value()` function but set the flag to false so that no escaping is done for them.